website/common/script/libs/taskDefaults.js

```diff
@@ -76,7 +76,10 @@ module.exports = function taskDefaults (task = {}) {
         s: true,
         su: true,
       },
-      startDate: moment().startOf('day').toDate(),
+      // If cron will happen today, start the daily yesterday
+      startDate: moment().startOf('day').add(user.preferences.dayStart, 'hour').isAfter(moment()) ?
```

Comment:
I think there are two changes to make here:

- it should also use the user's timezone to determine when the start of the day is (user.preferences.timezoneOffset)
- `moment().startOf('day')` should be cached